### PR TITLE
SDIT-2010: ♻️ Rename hospital service as will be returning prisons too

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: server/routes/addPatient/addPatientCompleted.test.ts [hunk 1]
```diff
@@ -2,32 +2,33 @@ import { Express } from 'express'
...
     })
 
-    hospitalSearchService.getHospital.mockResolvedValue({
+    agencySearchService.getAgency.mockResolvedValue({
       agencyId: 'SHEFF',
       description: 'Sheffield Hospital',
       longDescription: 'Sheffield Teaching Hospital',
       agencyType: 'HOSP',
       active: true,
-    } as Hospital)
```
Comment (petergphillips) [inline server/routes/addPatient/addPatientCompleted.test.ts:30]: this is an odd one - the service has always returned a `Prison`

File: server/services/agencySearchService.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      client.getAgenciesByType('HOSPITAL'),
+      client.getAgenciesByType('HSHOSP'),
+    ])
+
+    return [...hospitalType, ...hshospType]
+      .filter(h => h.active)
+      .sort((a: Prison, b: Prison) => a.description.localeCompare(b.description))
+  }
+
+  async getPrisons(user: Context): Promise<Prison[]> {
```
Comment (petergphillips) [inline server/services/agencySearchService.ts:20]: added new `getPrisons` method for returning prisons too

File: server/routes/addPatient/addPatientCompleted.test.ts (hunk 1 above)
Comment (petergphillips) [reply server/routes/addPatient/addPatientCompleted.test.ts:30]: although I guess it should actually be an `Agency`, will do in next PR...

Comment (mikehalmamoj) [reply server/routes/addPatient/addPatientCompleted.test.ts:30]: Not sure I get it - wouldn't agencyType by INST if this was returning a prison?

**Review by andymarke — APPROVED** [review]
lgtm

**Pull request merged**